Source/WebCore/dom/ViewTransition.cpp

```diff
@@ -856,8 +856,20 @@ LayoutRect ViewTransition::containingBlockRect()
     RefPtr frameView = document->view();
     if (!frameView)
         return { };
-    // FIXME: Bug 285400 - Correctly account for insets.
-    return { LayoutPoint { }, frameView->visibleContentRectIncludingScrollbars().size() };
+
+    LayoutRect result = frameView->boundsRect();
+
+    FloatBoxExtent insets;
+    if (RefPtr page = document->page()) {
+#if PLATFORM(IOS_FAMILY)
+        insets = page->obscuredInsets();
+#else
+        insets = page->obscuredContentInsets();
+#endif
+    }
+
+    result.moveBy({ LayoutUnit::fromFloatRound(-insets.left()), LayoutUnit::fromFloatRound(-insets.top()) });
+    return LayoutRect(encloseRectToDevicePixels(result, document->deviceScaleFactor()));
```

Comment:
what's the idea behind  pixel-snapping as if we were painting and convert (implicit floor?) to a type that's used for layout?